pkg/apis/certmanager/v1/types_issuer.go

```diff
@@ -277,6 +277,12 @@ type CAIssuer struct {
 	// If not set, certificates will be issued without distribution points set.
 	// +optional
 	CRLDistributionPoints []string `json:"crlDistributionPoints,omitempty"`
+
+	// The OCSP server is an X.509 v3 extension that defines the url of the OCSP responder.
+	// The OCSP responder can be queried for the revocation status of an issued certificate.
+	// If not set the certificate wil be issued without the OCSP server set.
+	// +optional
+	OCSPServer []string `json:"ocspServer,omitempty"`
```

Comment:
This name is singular, but it's a slice? Should this be `ocspServers`?

Reply:
Aaah I know why it was using a singular name: the standard library's [x509.go](https://github.com/golang/go/blob/fca94ab3ab113ceddb7934f76d0f1660cad98260/src/crypto/x509/x509.go#L701-L703) uses a singular name!

```go
// RFC 5280, 4.2.2.1 (Authority Information Access)
OCSPServer            []string
IssuingCertificateURL []string
```

The [RFC 5280](https://tools.ietf.org/html/rfc5280#section-4.2.2.1) describe the authority information access field (= analogous to our ocspServer array) as (ASN.1-like syntax):

```
   id-pe-authorityInfoAccess OBJECT IDENTIFIER ::= { id-pe 1 }

   AuthorityInfoAccessSyntax  ::=
           SEQUENCE SIZE (1..MAX) OF AccessDescription

   AccessDescription  ::=  SEQUENCE {
           accessMethod          OBJECT IDENTIFIER,
           accessLocation        GeneralName  }

   id-ad OBJECT IDENTIFIER ::= { id-pkix 48 }

   id-ad-caIssuers OBJECT IDENTIFIER ::= { id-ad 2 }

   id-ad-ocsp OBJECT IDENTIFIER ::= { id-ad 1 }
```

Should I still change `OCSPServer` to `OCSPServers`?

Reply:
Yes, i have reused the naming from `x509.go` and the certificate.

Reply:
We did this previously (reused the names from x509 for some fields) and eventually moved away from it (e.g. 'organization -> organizations').

I think `OCSPServers` makes sense to me! 😄